hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -808,8 +813,20 @@ void setFinalResult(Throwable failure) {
     }
 
     void updateQuorumSize(int newQuorumSize) {
-        synchronized (lock) {
-            jobRecord = jobRecord.withQuorumSize(newQuorumSize);
+        if (jobRecord.getQuorumSize() < newQuorumSize) {
+            jobRecord.setQuorumSize(newQuorumSize);
+            writeJobRecord(true);
+            logger.info("Current quorum size: " + jobRecord.getQuorumSize() + " of job "
+                    + idToString(jobRecord.getJobId()) + " is updated to: " + newQuorumSize);
+        }
+    }
+
+    private void writeJobRecord(boolean async) {
```

Comment:
it's better to split this into two methods and call the method updateJobRecord/updateJobRecordAsync I think